crates/runtime/src/http/v1/query.rs

```diff
@@ -102,18 +105,379 @@ use super::{sql_to_http_response, ArrowFormat};
+(default), 'text/csv' or 'text/plain'."),
+    ),
+    request_body(
+        content_type = "application/json",
+        description = "Prepared SQL query and its parameters to execute",
+        example = json!({
+            "sql": "SELECT $1, $2",
+            "params": [1, 7]
+        })
+    ),
+    responses(
+        (status = 200, description = "SQL query executed successfully (JSON format)", content((
+            Vec<serde_json::Value> = "application/json",
+            example = json!([
+                {
+                    "$1": 1,
+                    "$2": 7
+                }
+            ])
+        ),
+        (
+            String = "text/csv",
+            example = r#""$1","$2"
+1,7"#
+        ),
+        (
+            String = "text/plain",
+            example = r#"
+            +----+----+
+            | $1 | $2 |
+            +----+----+
+            | 1  | 7  |
+            +----+----+"#
+        ))),
+        (status = 400, description = "Invalid SQL query or malformed input", content((
+            String,
+            example = "Failed to execute query: Error during planning: table not found"
+        ))),
+        (status = 400, description = "Invalid SQL query or malformed input", content((
+            String,
+            example = "Failed to parse the request body as JSON: EOF while parsing an object"
+        ))),
+        (status = 500, description = "Internal server error", content((
+            String,
+            example = "Unexpected internal server error occurred"
+        )))
+        )
+    )
+)]
+pub(crate) async fn prepared(
+    Extension(df): Extension<Arc<DataFusion>>,
+    accept: Option<TypedHeader<Accept>>,
+    Json(PreparedStatement { sql, params }): Json<PreparedStatement>,
+) -> Response {
+    let params = match convert_json_to_param_values(params) {
+        Ok(params) => params,
+        Err(e) => {
+            tracing::debug!("Error reading params: {e}");
+            return (StatusCode::BAD_REQUEST, e.to_string()).into_response();
+        }
+    };
+
+    sql_to_http_response(
+        df,
+        &sql,
+        Some(params),
+        ArrowFormat::from_accept_header(accept.as_ref()),
+    )
+    .await
+}
+
+/// Converts a serde_json::Value into a datafusion::common::ParamValues.
+///
+/// # Arguments
+/// * `json` - The JSON value to convert.
+///
+/// # Returns
+/// * `Result<ParamValues, String>` - The converted ParamValues or an error message.
+///
+/// # Supported JSON Formats
+/// - Array: Converted to positional parameters (ParamValues::Positional).
+/// - Object: Converted to named parameters (ParamValues::Named).
+/// - Primitive types (null, bool, number, string) are supported within arrays or objects.
+///
+/// # Errors
+/// - Returns an error if the top-level JSON is not an array or object.
+/// - Returns an error if any value cannot be converted to a ScalarValue.
+fn convert_json_to_param_values(json: Value) -> Result<ParamValues, String> {
```

Comment:
Let's use a Snafu error instead of a String error, see https://github.com/spiceai/spiceai/blob/trunk/docs/dev/style_guide.md and https://github.com/spiceai/spiceai/blob/trunk/docs/dev/error_handling.md